### ISSUE TITLE
Feature: Add function for retrieving active process event

### ISSUE THREAD

**Issue by phemmer** [timescale/timescaledb#2675]
Currently when you do `SELECT wait_event, wait_event_type FROM pg_stat_activity`, you get back `Extension, Extension` if the process is currently executing within the TimescaleDB extension.

This lack of visibility into the extension state makes troubleshooting difficult, as the extension could be doing any number of things. This is even more important in multi-node configuration as when identifying bottlenecks, the extension could be processing on the access node, connecting to a data node, reading/writing to one or more of the data nodes, etc.

As such I think there should be a function added which can return the state of the extension inside a given pid. Something that would allow queries such as:

    select * from pg_stat_activity a, timescaledb_stat_activity(a.pid) b;

Basically any time `wait_event='Extension'`, the `timescaledb_stat_activity()` should return some meaningful representation of the extension's state. If the PID is not running inside the extension, it should return null values.

Also in the event that the access node is waiting on a data node, the state should reflect which data node. Since this does mean that different data nodes might be in different states, this might require that one of the columns returned by the function be an unstructured data type, like json.
